pysr/sr.py

```diff
@@ -2586,3 +2583,190 @@ def _mutate_parameter(param_name: str, param_value):
+            Each element is how to weight the mean-square-error loss
+            for that particular element of `X`. Alternatively,
+            if a custom `loss` was set, it will can be used
+            in arbitrary ways.
+        variable_names : list[str]
+            A list of names for the variables, rather than "x0", "x1", etc.
+            If `X` is a pandas dataframe, the column names will be used
+            instead of `variable_names`. Cannot contain spaces or special
+            characters. Avoid variable names which are also
+            function names in `sympy`, such as "N".
+            The number of variable names must be equal to recurrence_history_length.
+        X_units : list[str]
+            A list of units for each variable in `X`. Each unit should be
+            a string representing a Julia expression. See DynamicQuantities.jl
+            https://symbolicml.org/DynamicQuantities.jl/dev/units/ for more
+            information.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+        # Init attributes that are not specified in BaseEstimator
+        if self.warm_start and hasattr(self, "julia_state_stream_"):
+            pass
+        else:
+            if hasattr(self, "julia_state_stream_"):
+                warnings.warn(
+                    "The discovered expressions are being reset. "
+                    "Please set `warm_start=True` if you wish to continue "
+                    "to start a search where you left off.",
+                )
+
+            self.equations_ = None
+            self.nout_ = 1
+            self.selection_mask_ = None
+            self.julia_state_stream_ = None
+            self.julia_options_stream_ = None
+            self.complexity_of_variables_ = None
+            self.X_units_ = None
+            self.y_units_ = None
+
+        self._setup_equation_file()
+
+        runtime_params = self._validate_and_modify_params()
+
+        if self.recursive_history_length <= 0:
+            raise ValueError(
+                "The `recursive_history_length` must be greater than 0 (otherwise it's not recursion)."
+            )
+        if 1 not in X.shape and len(X.shape) > 1:
+            raise ValueError(
+                "Recursive symbolic regression requires a single input variable; reshape the array with array.reshape(-1, 1) or array.reshape(1, -1)"
+            )
+        if len(X) <= self.recursive_history_length + 1:
+            raise ValueError(
+                f"Recursive symbolic regression with a history length of {self.recursive_history_length} requires at least {self.recursive_history_length + 2} datapoints."
+            )
+        y = X.copy()
+        X = []
+        for i in range(self.recursive_history_length, len(y)):
+            X.append(y[i - self.recursive_history_length : i].flatten())
+        X = np.array(X)
```

Comment:
This should also be used in a `def predict(self, ...):` method to construct the input data for `super().predict(`